src/__tests__/middleware.test.ts

```diff
@@ -58,15 +90,53 @@ describe('Request mutation', () => {
       expect(event.headers[key]).toBe(value);
     }
   });
+  it('returns early if middleware returns a result', async () => {
+    const event = createEvent(gql`
+      query {
+        hello
+      }
+    `);
+    const lambdaHandler = startServerAndCreateLambdaHandler(
+      server,
+      handlers.createAPIGatewayProxyEventV2RequestHandler(),
+      {
+        context: async () => {
+          return {
+            foo: null,
+          };
+        },
+        middleware: [
+          async (event) => {
+            event.headers[''];
```

Comment:
Stray / leftover code?
```suggestion
          async (_event) => {
```